packages/block-editor/src/store/test/reducer.js

```diff
@@ -22,7 +22,11 @@ import {
 	blocks,
 	isTyping,
 	isCaretWithinFormattedText,
-	blockSelection,
+	selection,
+	selectionStart,
+	selectionEnd,
+	initialPosition,
+	isMultiSelecting,
```

Comment:
These flattened reducers have some overlap in the types of actions to which they react. One thing that worries me then is that, by only testing them as independent reducers, we may be omitting useful tests that look at the compound state of these reducers.

Could we keep this state compound while still benefitting from the simplicity of flattened reducers? What if, in `reducers`, we still write `selectionStart` _et al._ as independent functions, but at the end combine them as `blockSelection: combineReducers( { selectionStart, selectionEnd, initialPosition, isMultiSelecting } )`? Or is this just completely against the objectives behind this PR?

Reply:
What's the overlay? `selection` and `selectionStart`/`selectionEnd` don't have any overlap. `isMultiSelecting` doesn't have any overlap with the others either. `initialPosition` has a very limited overlap, where the state is very different from the rest. I don't really see a problem here.